test/private.jl

```diff
@@ -43,6 +43,20 @@ end
     end
 end
 
+@kernel function forloop2(A, ::Val{N}) where N
+    I = @index(Global, Linear)
+    i = @index(Local, Linear)
+    @private priv = ntuple(j -> A[i,j], N)
```

Comment:
This is failing on the GPU:
https://buildkite.com/julialang/kernelabstractions-dot-jl/builds/83#549a480a-7b87-4da0-b86a-1f03e9a3e76b
```
InvalidIRError: compiling kernel gpu_forloop2(Cassette.Context{nametype(CUDACtx),KernelAbstractions.CompilerMetadata{DynamicSize,DynamicCheck,Nothing,CartesianIndices{1,Tuple{Base.OneTo{Int64}}},NDRange{1,DynamicSize,DynamicSize,CartesianIndices{1,Tuple{Base.OneTo{Int64}}},CartesianIndices{1,Tuple{Base.OneTo{Int64}}}}},Nothing,KernelAbstractions.var"##PassType#253",Nothing,Cassette.DisableHooks}, typeof(gpu_forloop2), CuDeviceArray{Int64,2,1}, Val{64}) resulted in invalid LLVM IR
```

Reply:
`Reason: unsupported use of an undefined name (use of 'N')` huh